File: java/trip-service-kata/src/test/java/org/craftedsw/tripservicekata/user/UserShould.java
package org.craftedsw.tripservicekata.user;

import static org.hamcrest.core.Is.is;
import static org.craftedsw.tripservicekata.trip.UserBuilder.*;
import static org.junit.Assert.*;

import org.junit.Test;

public class UserShould {

	private static final User VARYS = new User();
	private static final User DAENERYS = new User();

	@Test
	public void inform_if_users_are_not_friends() throws Exception {
		User tyrion = aUser()
						.addFriend(DAENERYS)
						.build();
		
		assertThat(tyrion.isFriendWith(VARYS), is(false));
	}
	
	@Test
	public void inform_if_users_are_friends() throws Exception {
		User tyrion = aUser()
				.addFriend(DAENERYS)
				.addFriend(VARYS)
				.build();
		
		assertThat(tyrion.isFriendWith(VARYS), is(true));
	}
}

File: src/test/kotlin/tripservicekata/user/UserShould.kt
package org.craftedsw.tripservicekata.user

import org.junit.jupiter.api.Test
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class UserShould {

    private val tyrion = User()
    private val varys = User()
    private val cersei = User()

    @Test
    fun `notify if users are not friends`() {
        tyrion.addFriend(varys)

        assertFalse(tyrion.isFriendWith(cersei))
    }

    @Test
    fun `notify if users are friends`() {
        tyrion.addFriend(varys)
        tyrion.addFriend(cersei)

        assertTrue(tyrion.isFriendWith(varys))
    }
}
